docs/porting-guide.rst

```diff
@@ -2480,12 +2483,11 @@ Function : plat\_crash\_console\_flush
 ::
 
     Argument : void
-    Return   : int
+    Return   : void
```

Comment:
> I think the code should definitely always run through all consoles even if some of them fail, so the question is just what we want to return in the failure case. We could either return -1 if at least one console fails or return -1 only if all consoles fail.

I already said what I think - the former.

> I really can't come up with a scenario where either of these results would be terribly useful for a caller and allow it to make an important decision. Consider that almost all characters output by Trusted Firmware go through tf_printf() anyway, which already throws the return code of putchar() away and returns void.

Not all TF code is upstream and some of it may be used by other projects. Although the error code is not useful in current upstream client code, making a unilateral decision to remove it is unnecessary and off-topic.

> I think we just have a very different situation from what the stdio.h API was designed for (where output can be piped into files or devices and errors like "out of space" are common and meaningful) and keeping this functionality isn't worth all the extra effort and complexity it requires.

Who's to say the code may not get used in future in the way you describe (for piping into devices/files). Keeping this functionality doesn't seem much effort to me (an extra register save), and dealing with the compatibility issues is more effort.